### ISSUE TITLE
Attacking while in basic Ambush stance denies enemy reinforcements while still allowing yours.

### ISSUE THREAD

**Issue by Bogdanov89** [chadvandy/cbfm_wh3#274]
**Describe the bug**
Attacking an enemy while you are moving in basic Ambush stance denies their reinforcements, while still allowing yours.
This allows the basic ambush stance to work like a super powerful version of the old TWW2 Lightning Strike.
Best shown in a video at this reddit link https://www.reddit.com/r/totalwar/comments/xc3uui/attacking_an_enemy_while_you_are_in_ambush_stance/

**Expected behavior**
Basic ambush stance should behave the same as regular walking stance when it is used to move & attack armies on your turn.

**To Reproduce**
1. Play as any faction with basic ambush stance
2. Get 2 armies and attack 2 enemy armies that are within reinforcement range of each other
3. (optional) Choose normal walking stance & attack, then notice both you and enemy get reinforcements (works properly).
4. Choose ambush stance & attack (same conditions), then notice you get reinforcements while enemy does not.

**Screenshots**
Video available at https://www.reddit.com/r/totalwar/comments/xc3uui/attacking_an_enemy_while_you_are_in_ambush_stance/

**Additional context**
Basic ambush stance when moving should not influence the presence of reinforcements, since the basic ambush stance is supposed to do nothing when your army is moving in it.
If a proper fix is not possible via mods then perhaps just disabling the capability to attack while moving in basic ambush stance is good enough (just like you can not attack from encamp stance but can move in it).

**Comment by mfvreeland** [comment]
You can attack in ambush stance? That in and of itself seems like a bug if true.

**Comment by Bogdanov89** [comment]
Yes you can issue an attack command while in basic ambush stance, since CA put in the new QOL feature that you dont have to exit a stance to move a bit to just re-enter the stance.

However (unlike Encamp stance) the basic Ambush stance allows you to attack an enemy lord/city (in addition to moving).
This way of using the basic Ambush stance will still have the benefits of denying enemy any nearby reinforcing armies while at the same time allowing your own reinforcements.

Its basically just as if the enemy walked into your ambush (traditional tww1/tww2 style), but in Tww3 you can move and click to attack enemies while in basic ambush stance.

**Comment by mfvreeland** [comment]
Attacking in ambush stance is supposed to be a special feature of Skaven and Beastmen. I would say it is 100% a bug that this is allowed for all races. I don't know if this is fixable, but it might be, so I'll look into it.

**Comment by Bogdanov89** [comment]
From the latest 2.3.0 Tww3 patch notes:
* Attacking an army while in the Ambush stance will no longer prevent your opponent from bringing reinforcements.

Hopefully the CA fix will function as advertised :)

**Comment by Harmandodelavega** [comment]
Pls don't post just to report patch notes, unless you have verified the bug is fixed

**Comment by mfvreeland** [comment]
Interesting that they decided to leave attacking in ambush stance on and just address the reinforcement problem directly. I will test this to be sure, but it seems like a straightforward fix.
